src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+     * @return
+     * @throws InforException
+     */
+    @Override
+    public String deleteWorkOrderEquipment(InforContext context, String parentWorkorderID, String equipmentID) throws InforException {
+        MECService.validateInput(parentWorkorderID, equipmentID);
+
+        MP7396_RemoveWorkOrderEquipment_001 mp7396_removeWorkOrderEquipment_001 = new MP7396_RemoveWorkOrderEquipment_001();
+
+        ORGANIZATIONID_Type organizationid_type = new ORGANIZATIONID_Type();
+        organizationid_type.setORGANIZATIONCODE("*");
+
+        WOID_Type woid_typeParent = new WOID_Type();
+        woid_typeParent.setJOBNUM(parentWorkorderID);
+        woid_typeParent.setORGANIZATIONID(organizationid_type);
+
+        WOID_Type woid_typeEquipment = new WOID_Type();
+        woid_typeEquipment.setJOBNUM(equipmentID);
+        woid_typeEquipment.setORGANIZATIONID(organizationid_type);
+
+        mp7396_removeWorkOrderEquipment_001.setWORKORDERID(woid_typeParent);
+        mp7396_removeWorkOrderEquipment_001.setRELATEDWORKORDERID(woid_typeEquipment);
+
+        tools.performInforOperation(context, inforws::removeWorkOrderEquipmentOp, mp7396_removeWorkOrderEquipment_001);
+
+        return "OK";
+    }
+
+    /**
+     * Returns a list of all the equipments of the target workorder.
+     *
+     * @param context     the user credentials
+     * @param workorderID the ID of the parent workorder
+     * @return            the list of equipments
+     * @throws InforException
+     */
+    @Override
+    public GridRequestResult getWorkOrderEquipmentsOfWorkorder(InforContext context, String workorderID) throws InforException {
+        MECService.validateInput(workorderID);
+
+        GridRequest gridRequest = new GridRequest("WSJOBS_MEC", GridRequest.GRIDTYPE.LIST, 50);
+        gridRequest.addParam("param.workordernum", workorderID);
+        gridRequest.addParam("param.organization", tools.getOrganizationCode(context));
+        gridRequest.addParam("param.workorderrtype", "BR");
+        gridRequest.addParam("param.tenant", tools.getTenant(context));
+
+        return gridsService.executeQuery(context, gridRequest);
+    }
+
+    /**
+     * Updates the target workorder with the specified properties.
+     *
+     * @param context           the user credentials
+     * @param parentWorkorderID the ID of the parent workorder
+     * @param equipmentID       the ID of the equipment to update
+     * @param mecProperties     the new properties of the equipment to update
+     * @return
+     * @throws InforException
+     */
+    @Override
+    public String syncWorkOrderEquipment(InforContext context, String parentWorkorderID, String equipmentID, MEC mecProperties) throws InforException {
+        MECService.validateInput(parentWorkorderID, equipmentID, mecProperties);
+
+//        MP73
+
+        GridRequestResult woList = this.getWorkOrderEquipmentsOfWorkorder(context, parentWorkorderID);
```

Comment:
The fact that this is a gridrequestresult should be hidden from every method except the one which is actually performing the read.